freqtrade/tests/exchange/test_exchange.py

```diff
@@ -163,7 +168,19 @@ def test_get_ticker(mocker, ticker):
     ticker = get_ticker(pair='BTC_ETH')
     assert ticker['bid'] == 0.00001098
     assert ticker['ask'] == 0.00001099
+
+    # if not caching the result we should get the same ticker
+    ticker = get_ticker(pair='BTC_ETH', refresh=False)
```

Comment:
This condition is not tested properly.  You should change the ticker before to see that get_ticker() still return the same.

The test could look like that.
```python
def test_get_ticker(mocker, ticker):

    api_mock = MagicMock()
    api_mock.get_ticker = MagicMock(return_value=ticker())
    mocker.patch('freqtrade.exchange._API', api_mock)

    ticker = get_ticker(pair='BTC_ETH')
    assert ticker['bid'] == 0.00001098
    assert ticker['ask'] == 0.00001099

    # if not caching the result we should get the same ticker
    api_mock.get_ticker = MagicMock(return_value={"bid": 0, "ask": 1})
    mocker.patch('freqtrade.exchange._API', api_mock)
    ticker = get_ticker(pair='BTC_ETH', refresh=False)
    assert ticker['bid'] == 0.00001098
    assert ticker['ask'] == 0.00001099

    # change the ticker
    api_mock.get_ticker = MagicMock(return_value={"bid": 2, "ask": 3})
    mocker.patch('freqtrade.exchange._API', api_mock)

    ticker = get_ticker(pair='BTC_ETH', refresh=True)
    assert ticker['bid'] == 2
    assert ticker['ask'] == 3
```

However, when I execute this test, it fails.

```bash
22:01 $ pytest freqtrade/tests/exchange/
============================= test session starts ==============================
platform darwin -- Python 3.6.3, pytest-3.3.1, py-1.5.2, pluggy-0.6.0
rootdir: /Users/geraldlonlas/Desktop/dev/freqtrade, inifile:
plugins: mock-1.6.3, cov-2.5.1
collected 21 items

freqtrade/tests/exchange/test_exchange.py ..............F...             [ 85%]
freqtrade/tests/exchange/test_exchange_bittrex.py ...                    [100%]

=================================== FAILURES ===================================
_______________________________ test_get_ticker ________________________________

mocker = <pytest_mock.MockFixture object at 0x1073437b8>
ticker = {'ask': 1, 'bid': 0}

    def test_get_ticker(mocker, ticker):

        api_mock = MagicMock()
        api_mock.get_ticker = MagicMock(return_value=ticker())
        mocker.patch('freqtrade.exchange._API', api_mock)

        ticker = get_ticker(pair='BTC_ETH')
        assert ticker['bid'] == 0.00001098
        assert ticker['ask'] == 0.00001099

        # if not caching the result we should get the same ticker
        api_mock.get_ticker = MagicMock(return_value={"bid": 0, "ask": 1})
        ticker = get_ticker(pair='BTC_ETH', refresh=False)
>       assert ticker['bid'] == 0.00001098
E       assert 0 == 1.098e-05

freqtrade/tests/exchange/test_exchange.py:175: AssertionError
===================== 1 failed, 20 passed in 0.19 seconds ======================
```